quil-rs/src/program/scheduling/graphviz_dot.rs

```diff
@@ -100,91 +140,182 @@ impl<'a> ScheduledProgram<'a> {
+                                            next_block.label(),
+                                            *next_index,
+                                        )
+                                        .to_string(),
+                                    ),
                                 )
-                                .as_str(),
+                                .attributes()
+                                .set_label(
+                                    format!("if {} != 0", condition.to_quil_or_debug(),).as_str(),
+                                );
+                        };
+                    }
+                    BasicBlockTerminator::JumpUnless(JumpUnless { target, condition }) => {
+                        digraph
+                            .edge(
+                                &terminator_source_label,
+                                get_node_id(
+                                    &ScheduledGraphNode::BlockStart,
+                                    &DotFormatBlockLabel::from(target).to_string(),
+                                ),
+                            )
+                            .attributes()
+                            .set_label(
+                                format!("if {} != 0", condition.to_quil_or_debug(),).as_str(),
                             );
-                        if let Some(next_label) = next_block_label {
+                        if let Some((next_index, next_block)) = next_index_and_block {
                             digraph
                                 .edge(
                                     &terminator_source_label,
-                                    get_node_id(&ScheduledGraphNode::BlockStart, &next_label),
+                                    get_node_id(
+                                        &ScheduledGraphNode::BlockStart,
+                                        &DotFormatBlockLabel::from_label_or_index(
+                                            next_block.label(),
+                                            *next_index,
+                                        )
+                                        .to_string(),
+                                    ),
                                 )
                                 .attributes()
                                 .set_label(
-                                    format!(
-                                        "if {} {} 0",
-                                        condition.to_quil_or_debug(),
-                                        equality_operators.1
-                                    )
-                                    .as_str(),
+                                    format!("if {} == 0", condition.to_quil_or_debug(),).as_str(),
                                 );
                         };
                     }
-                    BlockTerminator::Unconditional { target } => {
+                    // BasicBlockTerminator::Conditional {
```

Comment:
todo: I want to come back to this and rework `BasicBlockTerminator::JumpWhen` and `::JumpUnless` into a single enum similar to this previous `Conditional`.

Reply:
Should that be considered a future `todo` or something to be done as part of #334?